apps/coordinator/ts/redis/__tests__/redis.service.test.ts

```diff
@@ -0,0 +1,122 @@
+import { ESupportedNetworks } from "../../common";
+import { RedisService } from "../redis.service";
+import { IScheduledPoll } from "../types";
+import { getPollKeyFromObject } from "../utils";
+
+const REDIS__GET_ALL_PREFIX = "*-test";
+
+const scheduledPoll: IScheduledPoll = {
+  maciAddress: "0xb83074Ac11fc569AC12F1b7D0C0a6809c3dc355b",
+  pollId: "5",
+  mode: 1,
+  chain: ESupportedNetworks.OPTIMISM_SEPOLIA,
+  endDate: 1752534000,
+};
+
+describe("RedisService", () => {
+  let service: RedisService;
+
+  beforeAll(async () => {
+    service = new RedisService();
+    await service.onModuleInit();
+    // Clean up before tests
+    const keys = await service.client.keys(REDIS__GET_ALL_PREFIX);
+    if (keys.length) {
+      await service.client.del(keys);
+    }
+  });
+
+  afterAll(async () => {
+    // Clean up after tests
+    const keys = await service.client.keys(REDIS__GET_ALL_PREFIX);
+    if (keys.length) {
+      await service.client.del(keys);
+    }
+
+    await service.client.quit();
+  });
+
+  it("should connect to Redis without issues", () => {
+    expect(service.client.isOpen).toBe(true);
+  });
+
+  it("should set and get a value, and parse it as IScheduledPoll", async () => {
+    const key = getPollKeyFromObject(scheduledPoll, true);
+    const value = JSON.stringify(scheduledPoll);
+
+    await service.set(key, value);
+
+    const stored = await service.get(key);
+    expect(stored).not.toBeNull();
+
+    const parsed = JSON.parse(stored!) as IScheduledPoll;
+
+    expect(parsed).toEqual(scheduledPoll);
+  });
+
+  it("should return null when non-existent key is requested", async () => {
+    const nonExistentKey = "non-existent-key";
+    const result = await service.get(nonExistentKey);
+
+    expect(result).toBeNull();
+  });
+
+  it("should set and get multiple values", async () => {
+    const pollOne = JSON.parse(JSON.stringify(scheduledPoll)) as IScheduledPoll;
```

Comment:
Nit: Do we need to stringify then parse every time? Could we just set pollId and then `stringify`?